ui/index.go

```diff
@@ -0,0 +1,82 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ui
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	cdnReact           = "https://cdnjs.cloudflare.com/ajax/libs/react/15.5.4/react.min.js"
+	cdnReactDom        = "https://cdnjs.cloudflare.com/ajax/libs/react/15.5.4/react-dom.min.js"
+	cdnBabelStandalone = "https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.24.0/babel.min.js"
+	cdnAxios           = "https://cdnjs.cloudflare.com/ajax/libs/axios/0.16.1/axios.min.js"
+)
+
+const indexHTML = `
+<!DOCTYPE HTML>
+<html>
+  <head>
+    <meta charset="utf-8">
+	<title>Vibrato Tech Test App</title>
+	<link rel="stylesheet" href="/css/site.css" type="text/css" />
+	<link href="https://fonts.googleapis.com/css?family=Arimo" rel="stylesheet">
+	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
+	<script defer src="https://use.fontawesome.com/releases/v5.0.6/js/all.js"></script>
+  </head>
+  <body>
+  	<header>
+    	<img src="/images/VIBRATO_Logo_dark-cropped-200.png" width="100" height="85"/>
+    </header>
+    <div id='root'></div>
+	<footer>
+        &COPY; Vibrato
+	</footer>
+	<script src="` + cdnReact + `"></script>
+    <script src="` + cdnReactDom + `"></script>
+    <script src="` + cdnBabelStandalone + `"></script>
+    <script src="` + cdnAxios + `"></script>
+	<script src="/js/app.jsx" type="text/babel"></script>
+  </body>
+</html>
+`
+
+func indexHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, indexHTML)
+	})
+}
+
+func assetHandler(cfg Config) http.Handler {
+	// so not secure!
+	return http.FileServer(cfg.Assets)
+}
+
+func uiHandler(cfg Config, router *mux.Router) {
+	router.PathPrefix("/js/").Handler(assetHandler(cfg))
+	router.PathPrefix("/css/").Handler(assetHandler(cfg))
+	router.PathPrefix("/images/").Handler(assetHandler(cfg))
+	router.PathPrefix("/swagger/").Handler(assetHandler(cfg))
```

Comment:
Why did you move these into a separate file, while leaving the health check where it is? Makes more sense to me to leave the global routes in`ui.go`, while index specific entries in `index.go`

Reply:
I was thinking of all these routes as ui/frontend routes as separate from the api routes. But the main file is already called ui, so I couldn't call it that. Healthcheck is neither frontend nor backend so stayed in the main.

I don't mind though, i'll move them back.